### PR TITLE
Documentation Updates

### PR DESCRIPTION
**Pull request by rhyshort**
## What
Updated the documentation to make it easier for contributors to get started.

## How
- Updated Contributing to include information on test settings.
- Added Issue and PR templates to ask for the required information.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -70,6 +70,24 @@ structure. All production code goes into `Source` and test code goes into `Test`
 * `Source` for classes that need to be used in order to successfully
 interact with the database. Such as CDTCouchDBClient.
 
+__NOTE__: Due to the lack of xcode support for Swift Package Manager source
+layouts, the `Source` directly cannot contain folders so the `swift build` command
```
Comment (brynh) [inline CONTRIBUTING.md]: Should this say Source `directory` rather than `directly`? Otherwise I really don't understand this sentence.

Comment (rhyshort) [inline CONTRIBUTING.md]: The bullets refer to the xcode project structure. Also fixed directly in dabbde2

**Comment by brynh** [comment]
+1

**Comment by emlaver** [comment]
@rhyshort Does the `build` need to be removed in this line?
https://github.com/cloudant/swift-cloudant/pull/78/files#diff-6a3371457528722a734f3c51d9238c13R97

I remember we chatted over Slack when I was having difficulties testing and you sent me an updated command:
```
xcodebuild -workspace SwiftCloudant.xcworkspace -scheme SwiftCloudant -destination 'platform=iOS Simulator,OS=latest,name=iPhone 6S' test 
```

**Comment by rhyshort** [comment]
Yes it should have been, ( I thought I had already done it, must have been a dream), anyway really done in 0b0dca7

**Comment by emlaver** [comment]
+1

**Pull request merged**